### PR TITLE
fixed dai/usdt pool issue

### PR DESCRIPTION
**Pull request by rango527**

### CODE REVIEW

File: packages/client/src/components/add-liquidity/add-liquidity-v3.tsx [hunk 1]
```diff
@@ -404,6 +404,13 @@ export const AddLiquidityV3 = ({
             }
         }
 
+        if (Math.abs(upperBoundTick - lowerBoundTick) == uniPool.tickSpacing) {
+            upperBoundTick +=
+                upperBoundTick > lowerBoundTick
```
Comment (zmanian) [inline packages/client/src/components/add-liquidity/add-liquidity-v3.tsx:409]: The upper bound tick should always be greater than than the lower bound tick. We should be bailing out earlier in the function if that isn't true.

**Review by zmanian — CHANGES_REQUESTED** [review]

File: packages/client/src/components/add-liquidity/add-liquidity-v3.tsx (hunk 1 above)
Comment (rango527) [reply packages/client/src/components/add-liquidity/add-liquidity-v3.tsx:409]: In this code the upperBoundTick is the tick calculated by upperBound, and the lowerBoundTick is the tick calculated by lowerBound.
And we check what tick is lower when call Position.fromAmounts.
image

Comment (zmanian) [reply packages/client/src/components/add-liquidity/add-liquidity-v3.tsx:409]: Yeah I think having a code path where the upper tick moves under the lower tick can lead to confusing outcomes

Comment (rango527) [reply packages/client/src/components/add-liquidity/add-liquidity-v3.tsx:409]: Yeah, it is a lilttle confusing.
Below that code, there are the code to calculate priceLower and priceUpper.
So we will have lowerBound, lowerBoundTick, priceLower as one pair, same for upper.
When we have infinity errors at last time, ticks were sorted and priceLower was calculated by lower tick sorted in ticks.
It was wrong and it causes errors.
The priceLower should be calculated by the tick of lowerBound, not by the lower tick in sorted ticks.

**Pull request merged**